geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/functions/GatewaySenderFunctionArgs.java

```diff
@@ -47,6 +47,7 @@
   private final List<String> gatewayEventFilters;
   private final List<String> gatewayTransportFilters;
   private final Boolean enforceThreadsConnectSameReceiver;
+  private final String state;
```

Comment:
The GatewaySenderFunctionArgs defines a state field. That object is only used by the CreateGatewaySenderCommand. The CreateGatewaySenderCommand doesn't have any way to set the state when a sender is created. Should it?

Reply:
Thank you for the comment!

Currently, `CreateGatewaySenderCommand` can only manipulate with the initial state of gateway-sender by using deprecated `--manual-start` parameter (default value is false). This parameter does not update the new `state` parameter.

As you suggested, we could also add `--state` parameter in `create gateway-sender`. We could maybe then completely remove the depreciated `--manual-start` parameter, since `--state` will do the same thing (and even more considering "paused" state) as `--manual-start` parameter. As it is now, these two parameters have functional conflict which is currently resolved as described in RFC under "Solution" chapter: https://cwiki.apache.org/confluence/display/GEODE/Persist+gateway-sender+state+within+Cluster+Configuration?moved=true

What do you think? Do you see benefit in having --state parameter in create gateway-sender command?

Reply:
One of the problems with manual-start is that if it is set to true, it doesn't recover the sender's persistence files.

The same is also true of state=stopped.

This causes data regions to not be recovered in the parallel case since the data and queue regions are co-located.

A warning like this is displayed:
```
[warn 2021/04/01 09:32:47.877 PDT server-ln-1 <ColocationLogger for Trade> tid=0x3c] Persistent data recovery for region /Trade is prevented by offline colocated region
	/ny_PARALLEL_GATEWAY_SENDER_QUEUE
```
A test to cause this is:

1. Start a server with a persistent region and sender
2. Shutdown the server
3. Set manual-start=true (either export / cluster configuration or modify xml)
4. Restart the server

If we allow a persistent, parallel sender to be created with state=stopped or even save the stopped state when the sender shuts down, we'll have the same problem. I'm not sure we should allow either of those cases.

Reply:
Thank you for pointing this out. I have reproduced the problem by following your steps. 

It seem that this PR is missing some cases with persistent (and also redundant) geateway-sender queues and region. I will create those cases and investigate this particular issue a little bit more, in order to see what can be done here.

Reply:
Hi @boglesby ,

I have covered the scenario when persistent parallel gateway sender is started in "stopped" state in following commit: https://github.com/apache/geode/pull/6036/commits/6abcb7aa7a027187afe64037e2a78e011f126d29

I only followed the similar scenario when gateway-sender is started, and then stopped using gfsh commands. In that case all relevant threads that handle events are stopped (no data is enqueued), but collocated queue and data regions continue coexist and work. So in order to cover this problematic case, I only bring gateway-sender to the same state as it would be after it is started and then stopped by using gfsh commands (start and stop gateway-sender).

You can find more information in below TC's that verify the solution. I tried to describe every TC in comment block just before the each case, so it would be easier to review. Hope this will help.

- ParallelGatewaySenderOperationClusterConfigDUnitTest.java
- ParallelGatewaySenderOperationClusterConfigDUnitTest.java

I would be really grateful if you could share your opinion on this solution. Also, I just noticed that I have introduced the changes you proposed in you first two comments. I will introduce them in next commit.